File: spec/java/src/io/kaitai/struct/spec/TestProcessToUser.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.ProcessToUser;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestProcessToUser extends CommonSpec {
    @Test
    public void testProcessToUser() throws Exception {
        ProcessToUser r = ProcessToUser.fromFile(SRC_DIR + "process_rotate.bin");

        assertEquals(r.buf1().str(), "Hello");
    }
}

File: spec/rust/tests/test_process_to_user.rs
#![allow(dead_code)]
use std::fs;

extern crate kaitai;
use self::kaitai::*;
mod formats;
use formats::process_to_user::*;

#[test]
fn test_process_to_user() {
    let bytes = fs::read("../../src/process_rotate.bin").unwrap();
    let reader = BytesReader::new(&bytes);
    let mut r = ProcessToUser::default();

    if let Err(err) = r.read(&reader, None, Some(KStructUnit::parent_stack())) {

        panic!("{:?}", err);
    }
    assert_eq!(r.buf1().str(), "Hello");
}
